From 075219e58fd4068c64d4be91befa4c24d534b743 Mon Sep 17 00:00:00 2001
From: Faith Ekstrand <faith.ekstrand@collabora.com>
Date: Thu, 19 Oct 2023 07:27:45 -0500
Subject: [PATCH 582/834] nvk: Advertise shaderOutputLayer and
 shaderOutputViewportIndex

We already advertise the extension but it's moved into feature bits in
Vulkan 1.2 and we didn't have those turned on.

Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/25812>
---
 src/nouveau/vulkan/nvk_physical_device.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/nouveau/vulkan/nvk_physical_device.c b/src/nouveau/vulkan/nvk_physical_device.c
index c7a9faa5e45..309294f561d 100644
--- a/src/nouveau/vulkan/nvk_physical_device.c
+++ b/src/nouveau/vulkan/nvk_physical_device.c
@@ -240,6 +240,8 @@ nvk_get_device_features(const struct nv_device_info *info,
       .bufferDeviceAddress = true,
       .bufferDeviceAddressCaptureReplay = false,
       .bufferDeviceAddressMultiDevice = false,
+      .shaderOutputViewportIndex = info->cls_eng3d >= MAXWELL_B,
+      .shaderOutputLayer = info->cls_eng3d >= MAXWELL_B,
 
       /* Vulkan 1.3 */
       .robustImageAccess = true,
-- 
2.42.0

